### ISSUE TITLE
[Bug] Redis array response is wrongly translated

### ISSUE THREAD

**Issue by gupta-mukul** [aembke/fred.rs#147]
Redis version - 7.0.5
Platform - linux
Using Docker and/or Kubernetes - no
Deployment type - cluster

I have implemented a Redis module that made a custom command available in Redis. This command returns an array with a single value of type integer.

When I run the command on redis-cli, It is correctly displayed as an array with a single element.

```
127.0.0.1:6379> udsf.delrec recordID1 RealmID1 StorageID
1) (integer) 0
127.0.0.1:6379>
```
However, when executing this command through fred, RedisValue received is of type RedisValue::Integer (and not RedisValue::Array). It works when an array contains more than one element.

It seems like fred is translating an array with only one element to a basic type. Since this is a common response format that is being shared across multiple commands (with some commands filling in more than one element), this is breaking the application parsing of response.

**To Reproduce**
1. Run a command on Redis that returns an array with a single integer value.

**Logs**
```
running 1 test
[2023-05-24T11:46:23Z DEBUG fred::router::commands] fred-wOdhQ0ju6H: Initializing router with policy: Some(Constant { attempts: 0, max_attempts: 0, delay: 1000, jitter: 100 })
[2023-05-24T11:46:23Z TRACE fred::router::clustered] fred-wOdhQ0ju6H: Adding 0 commands to retry buffer.
[2023-05-24T11:46:23Z DEBUG fred::router::clustered] fred-wOdhQ0ju6H: Synchronizing cluster state.
[2023-05-24T11:46:23Z DEBUG fred::router::clustered] fred-wOdhQ0ju6H: Attempting clustered connections to any of {Server { host: "127.0.0.1", port: 6379, tls_server_name: None }}
[2023-05-24T11:46:23Z TRACE fred::protocol::connection] fred-wOdhQ0ju6H: Checking connection type. Native-tls: false, Rustls: false
[2023-05-24T11:46:23Z TRACE fred::utils] Using timeout: 60000 ms
[2023-05-24T11:46:23Z TRACE fred::protocol::types] fred-wOdhQ0ju6H: Using 127.0.0.1 among 1 possible socket addresses for 127.0.0.1:6379
[2023-05-24T11:46:23Z DEBUG fred::protocol::connection] fred-wOdhQ0ju6H: Creating TCP connection to 127.0.0.1 at 127.0.0.1:6379
[2023-05-24T11:46:23Z TRACE fred::utils] Using timeout: 60000 ms
[2023-05-24T11:46:23Z TRACE fred::protocol::connection] fred-wOdhQ0ju6H: Skip authentication without credentials.
[2023-05-24T11:46:23Z DEBUG fred::protocol::connection] fred-wOdhQ0ju6H: Setting client name.
[2023-05-24T11:46:23Z TRACE fred::protocol::codec] fred-wOdhQ0ju6H: Encoded 51 bytes to 127.0.0.1:6379. Buffer len: 51 (RESP2)
[2023-05-24T11:46:23Z TRACE fred::protocol::codec] fred-wOdhQ0ju6H: Recv 5 bytes from 127.0.0.1:6379 (RESP2).
[2023-05-24T11:46:23Z TRACE fred::protocol::codec] fred-wOdhQ0ju6H: Parsed 5 bytes from 127.0.0.1:6379
[2023-05-24T11:46:23Z DEBUG fred::protocol::connection] fred-wOdhQ0ju6H: Successfully set Redis client name.
[2023-05-24T11:46:23Z TRACE fred::protocol::codec] fred-wOdhQ0ju6H: Encoded 24 bytes to 127.0.0.1:6379. Buffer len: 24 (RESP2)
[2023-05-24T11:46:23Z TRACE fred::protocol::codec] fred-wOdhQ0ju6H: Recv 0 bytes from 127.0.0.1:6379 (RESP2).
[2023-05-24T11:46:23Z TRACE fred::protocol::codec] fred-wOdhQ0ju6H: Recv 5 bytes from 127.0.0.1:6379 (RESP2).
[2023-05-24T11:46:23Z TRACE fred::protocol::codec] fred-wOdhQ0ju6H: Parsed 5 bytes from 127.0.0.1:6379
[2023-05-24T11:46:23Z DEBUG fred::protocol::connection] fred-wOdhQ0ju6H: Read client ID: Number { data: 32, attributes: None }
[2023-05-24T11:46:23Z TRACE fred::protocol::codec] fred-wOdhQ0ju6H: Encoded 26 bytes to 127.0.0.1:6379. Buffer len: 26 (RESP2)
[2023-05-24T11:46:23Z TRACE fred::protocol::codec] fred-wOdhQ0ju6H: Recv 0 bytes from 127.0.0.1:6379 (RESP2).
[2023-05-24T11:46:23Z TRACE fred::protocol::codec] fred-wOdhQ0ju6H: Recv 667 bytes from 127.0.0.1:6379 (RESP2).
[2023-05-24T11:46:23Z TRACE fred::protocol::codec] fred-wOdhQ0ju6H: Parsed 667 bytes from 127.0.0.1:6379
[2023-05-24T11:46:23Z DEBUG fred::protocol::connection] fred-wOdhQ0ju6H: Read server version Some(Version { major: 7, minor: 0, patch: 5 })
[2023-05-24T11:46:23Z TRACE fred::protocol::connection] fred-wOdhQ0ju6H: Checking cluster info for 127.0.0.1:6379
[2023-05-24T11:46:23Z TRACE fred::protocol::codec] fred-wOdhQ0ju6H: Encoded 27 bytes to 127.0.0.1:6379. Buffer len: 27 (RESP2)
[2023-05-24T11:46:23Z TRACE fred::protocol::codec] fred-wOdhQ0ju6H: Recv 0 bytes from 127.0.0.1:6379 (RESP2).
[2023-05-24T11:46:23Z TRACE fred::protocol::codec] fred-wOdhQ0ju6H: Recv 320 bytes from 127.0.0.1:6379 (RESP2).
[2023-05-24T11:46:23Z TRACE fred::protocol::codec] fred-wOdhQ0ju6H: Parsed 320 bytes from 127.0.0.1:6379
[2023-05-24T11:46:23Z DEBUG fred::router::clustered] fred-wOdhQ0ju6H: Connected to 127.0.0.1:6379 (1/1)
[2023-05-24T11:46:23Z TRACE fred::protocol::codec] fred-wOdhQ0ju6H: Encoded 28 bytes to 127.0.0.1:6379. Buffer len: 28 (RESP2)
[2023-05-24T11:46:23Z TRACE fred::protocol::codec] fred-wOdhQ0ju6H: Recv 0 bytes from 127.0.0.1:6379 (RESP2).
[2023-05-24T11:46:23Z TRACE fred::protocol::codec] fred-wOdhQ0ju6H: Recv 97 bytes from 127.0.0.1:6379 (RESP2).
[2023-05-24T11:46:23Z TRACE fred::protocol::codec] fred-wOdhQ0ju6H: Parsed 97 bytes from 127.0.0.1:6379
[2023-05-24T11:46:23Z TRACE fred::router::clustered] fred-wOdhQ0ju6H: Recv CLUSTER SLOTS response: Array([Array([Integer(0), Integer(16383), Array([String("127.0.0.1"), Integer(6379), String("ae05803aa47f17ac7fcc22e81d04a94ff7408fbe"), Array([])])])])
[2023-05-24T11:46:23Z DEBUG fred::router::clustered] fred-wOdhQ0ju6H: Rebuilding cluster state from host: 127.0.0.1
[2023-05-24T11:46:23Z TRACE fred::protocol::cluster] fred-wOdhQ0ju6H: Skip modifying TLS hostnames.
[2023-05-24T11:46:23Z DEBUG fred::router::clustered] fred-wOdhQ0ju6H: Changing cluster connections: ClusterChange { add: [Server { host: "127.0.0.1", port: 6379, tls_server_name: None }], remove: [] }
[2023-05-24T11:46:23Z DEBUG fred::modules::inner] fred-wOdhQ0ju6H: No `on_cluster_change` listeners.
[2023-05-24T11:46:23Z DEBUG fred::router::clustered] fred-wOdhQ0ju6H: Connecting to cluster node 127.0.0.1:6379
[2023-05-24T11:46:23Z TRACE fred::protocol::connection] fred-wOdhQ0ju6H: Checking connection type. Native-tls: false, Rustls: false
[2023-05-24T11:46:23Z TRACE fred::utils] Using timeout: 60000 ms
[2023-05-24T11:46:23Z TRACE fred::protocol::types] fred-wOdhQ0ju6H: Using 127.0.0.1 among 1 possible socket addresses for 127.0.0.1:6379
[2023-05-24T11:46:23Z DEBUG fred::protocol::connection] fred-wOdhQ0ju6H: Creating TCP connection to 127.0.0.1 at 127.0.0.1:6379
[2023-05-24T11:46:23Z TRACE fred::utils] Using timeout: 60000 ms
[2023-05-24T11:46:23Z TRACE fred::protocol::connection] fred-wOdhQ0ju6H: Skip authentication without credentials.
[2023-05-24T11:46:23Z DEBUG fred::protocol::connection] fred-wOdhQ0ju6H: Setting client name.
[2023-05-24T11:46:23Z TRACE fred::protocol::codec] fred-wOdhQ0ju6H: Encoded 51 bytes to 127.0.0.1:6379. Buffer len: 51 (RESP2)
[2023-05-24T11:46:23Z TRACE fred::protocol::codec] fred-wOdhQ0ju6H: Recv 5 bytes from 127.0.0.1:6379 (RESP2).
[2023-05-24T11:46:23Z TRACE fred::protocol::codec] fred-wOdhQ0ju6H: Parsed 5 bytes from 127.0.0.1:6379
[2023-05-24T11:46:23Z DEBUG fred::protocol::connection] fred-wOdhQ0ju6H: Successfully set Redis client name.
[2023-05-24T11:46:23Z TRACE fred::protocol::codec] fred-wOdhQ0ju6H: Encoded 24 bytes to 127.0.0.1:6379. Buffer len: 24 (RESP2)
[2023-05-24T11:46:23Z TRACE fred::protocol::codec] fred-wOdhQ0ju6H: Recv 0 bytes from 127.0.0.1:6379 (RESP2).
[2023-05-24T11:46:23Z TRACE fred::protocol::codec] fred-wOdhQ0ju6H: Recv 5 bytes from 127.0.0.1:6379 (RESP2).
[2023-05-24T11:46:23Z TRACE fred::protocol::codec] fred-wOdhQ0ju6H: Parsed 5 bytes from 127.0.0.1:6379
[2023-05-24T11:46:23Z DEBUG fred::protocol::connection] fred-wOdhQ0ju6H: Read client ID: Number { data: 33, attributes: None }
[2023-05-24T11:46:23Z TRACE fred::protocol::codec] fred-wOdhQ0ju6H: Encoded 26 bytes to 127.0.0.1:6379. Buffer len: 26 (RESP2)
[2023-05-24T11:46:23Z TRACE fred::protocol::codec] fred-wOdhQ0ju6H: Recv 0 bytes from 127.0.0.1:6379 (RESP2).
[2023-05-24T11:46:23Z TRACE fred::protocol::codec] fred-wOdhQ0ju6H: Recv 667 bytes from 127.0.0.1:6379 (RESP2).
[2023-05-24T11:46:23Z TRACE fred::protocol::codec] fred-wOdhQ0ju6H: Parsed 667 bytes from 127.0.0.1:6379
[2023-05-24T11:46:23Z DEBUG fred::protocol::connection] fred-wOdhQ0ju6H: Read server version Some(Version { major: 7, minor: 0, patch: 5 })
[2023-05-24T11:46:23Z TRACE fred::protocol::connection] fred-wOdhQ0ju6H: Checking cluster info for 127.0.0.1:6379
[2023-05-24T11:46:23Z TRACE fred::protocol::codec] fred-wOdhQ0ju6H: Encoded 27 bytes to 127.0.0.1:6379. Buffer len: 27 (RESP2)
[2023-05-24T11:46:23Z TRACE fred::protocol::codec] fred-wOdhQ0ju6H: Recv 0 bytes from 127.0.0.1:6379 (RESP2).
[2023-05-24T11:46:23Z TRACE fred::protocol::codec] fred-wOdhQ0ju6H: Recv 320 bytes from 127.0.0.1:6379 (RESP2).
[2023-05-24T11:46:23Z TRACE fred::protocol::codec] fred-wOdhQ0ju6H: Parsed 320 bytes from 127.0.0.1:6379
[2023-05-24T11:46:23Z DEBUG fred::router::clustered] fred-wOdhQ0ju6H: Finish synchronizing cluster connections.
[2023-05-24T11:46:23Z DEBUG fred::modules::inner] fred-wOdhQ0ju6H: No `on_reconnect` listeners.
[2023-05-24T11:46:23Z DEBUG fred::router::commands] fred-wOdhQ0ju6H: Starting command processing stream...
[2023-05-24T11:46:23Z TRACE fred::protocol::codec] fred-wOdhQ0ju6H: Recv 0 bytes from 127.0.0.1:6379 (RESP2).
[2023-05-24T11:46:23Z DEBUG fred::commands::impls::server] fred-wOdhQ0ju6H: Unique primary nodes in split: [Server { host: "127.0.0.1", port: 6379, tls_server_name: None }]
[2023-05-24T11:46:23Z DEBUG fred::router::commands] fred-jKw91HNi2O: Initializing router with policy: Some(Constant { attempts: 0, max_attempts: 0, delay: 1000, jitter: 100 })
[2023-05-24T11:46:23Z DEBUG fred::router::centralized] fred-jKw91HNi2O: Initializing centralized connection.
[2023-05-24T11:46:23Z TRACE fred::protocol::connection] fred-jKw91HNi2O: Checking connection type. Native-tls: false, Rustls: false
[2023-05-24T11:46:23Z TRACE fred::utils] Using timeout: 60000 ms
[2023-05-24T11:46:23Z TRACE fred::protocol::types] fred-jKw91HNi2O: Using 127.0.0.1 among 1 possible socket addresses for 127.0.0.1:6379
[2023-05-24T11:46:23Z DEBUG fred::protocol::connection] fred-jKw91HNi2O: Creating TCP connection to 127.0.0.1 at 127.0.0.1:6379
[2023-05-24T11:46:23Z TRACE fred::utils] Using timeout: 60000 ms
[2023-05-24T11:46:23Z TRACE fred::protocol::connection] fred-jKw91HNi2O: Skip authentication without credentials.
[2023-05-24T11:46:23Z DEBUG fred::protocol::connection] fred-jKw91HNi2O: Setting client name.
[2023-05-24T11:46:23Z TRACE fred::protocol::codec] fred-jKw91HNi2O: Encoded 51 bytes to 127.0.0.1:6379. Buffer len: 51 (RESP2)
[2023-05-24T11:46:23Z TRACE fred::protocol::codec] fred-jKw91HNi2O: Recv 5 bytes from 127.0.0.1:6379 (RESP2).
[2023-05-24T11:46:23Z TRACE fred::protocol::codec] fred-jKw91HNi2O: Parsed 5 bytes from 127.0.0.1:6379
[2023-05-24T11:46:23Z DEBUG fred::protocol::connection] fred-jKw91HNi2O: Successfully set Redis client name.
[2023-05-24T11:46:23Z TRACE fred::protocol::codec] fred-jKw91HNi2O: Encoded 24 bytes to 127.0.0.1:6379. Buffer len: 24 (RESP2)
[2023-05-24T11:46:23Z TRACE fred::protocol::codec] fred-jKw91HNi2O: Recv 0 bytes from 127.0.0.1:6379 (RESP2).
[2023-05-24T11:46:23Z TRACE fred::protocol::codec] fred-jKw91HNi2O: Recv 5 bytes from 127.0.0.1:6379 (RESP2).
[2023-05-24T11:46:23Z TRACE fred::protocol::codec] fred-jKw91HNi2O: Parsed 5 bytes from 127.0.0.1:6379
[2023-05-24T11:46:23Z DEBUG fred::protocol::connection] fred-jKw91HNi2O: Read client ID: Number { data: 34, attributes: None }
[2023-05-24T11:46:23Z TRACE fred::protocol::codec] fred-jKw91HNi2O: Encoded 26 bytes to 127.0.0.1:6379. Buffer len: 26 (RESP2)
[2023-05-24T11:46:23Z TRACE fred::protocol::codec] fred-jKw91HNi2O: Recv 0 bytes from 127.0.0.1:6379 (RESP2).
[2023-05-24T11:46:23Z TRACE fred::protocol::codec] fred-jKw91HNi2O: Recv 667 bytes from 127.0.0.1:6379 (RESP2).
[2023-05-24T11:46:23Z TRACE fred::protocol::codec] fred-jKw91HNi2O: Parsed 667 bytes from 127.0.0.1:6379
[2023-05-24T11:46:23Z DEBUG fred::protocol::connection] fred-jKw91HNi2O: Read server version Some(Version { major: 7, minor: 0, patch: 5 })
[2023-05-24T11:46:23Z DEBUG fred::modules::inner] fred-jKw91HNi2O: No `on_reconnect` listeners.
[2023-05-24T11:46:23Z DEBUG fred::router::commands] fred-jKw91HNi2O: Starting command processing stream...
[2023-05-24T11:46:23Z TRACE fred::protocol::codec] fred-jKw91HNi2O: Recv 0 bytes from 127.0.0.1:6379 (RESP2).
connected successfully
[2023-05-24T11:46:23Z TRACE fred::interfaces] fred-wOdhQ0ju6H: Sending command udsf.delrec (0) to router.
[2023-05-24T11:46:23Z TRACE fred::utils] Using timeout: 2000 ms
[2023-05-24T11:46:23Z TRACE fred::router::commands] fred-wOdhQ0ju6H: Recv command: RouterCommand { kind: "Command", command: "udsf.delrec" }
[2023-05-24T11:46:23Z TRACE fred::protocol::command] fred-wOdhQ0ju6H: Pipeline check udsf.delrec: true
[2023-05-24T11:46:23Z DEBUG fred::router::clustered] fred-wOdhQ0ju6H: Writing command `udsf.delrec` to 127.0.0.1:6379
[2023-05-24T11:46:23Z TRACE fred::router::utils] fred-wOdhQ0ju6H: Writing udsf.delrec (0). Timed out: false, Force flush: false
[2023-05-24T11:46:23Z TRACE fred::router::utils] fred-wOdhQ0ju6H: Sending command udsf.delrec (0) to 127.0.0.1:6379
[2023-05-24T11:46:23Z TRACE fred::protocol::connection] Writing and flushing 127.0.0.1:6379
[2023-05-24T11:46:23Z TRACE fred::protocol::codec] fred-wOdhQ0ju6H: Encoded 66 bytes to 127.0.0.1:6379. Buffer len: 66 (RESP2)
[2023-05-24T11:46:23Z TRACE fred::router::commands] fred-wOdhQ0ju6H: Sent command to 127.0.0.1:6379. Flushed: true
[2023-05-24T11:46:23Z TRACE fred::protocol::codec] fred-wOdhQ0ju6H: Recv 8 bytes from 127.0.0.1:6379 (RESP2).
[2023-05-24T11:46:23Z TRACE fred::protocol::codec] fred-wOdhQ0ju6H: Parsed 8 bytes from 127.0.0.1:6379
[2023-05-24T11:46:23Z TRACE fred::router::clustered] fred-wOdhQ0ju6H: Parsing response frame from 127.0.0.1:6379
[2023-05-24T11:46:23Z TRACE fred::router::clustered] fred-wOdhQ0ju6H: Checking response to udsf.delrec (0)
[2023-05-24T11:46:23Z TRACE fred::router::clustered] fred-wOdhQ0ju6H: Handling clustered response kind: Respond
[2023-05-24T11:46:23Z TRACE fred::protocol::responders] fred-wOdhQ0ju6H: Respond to caller from 127.0.0.1:6379 for udsf.delrec with Array
[2023-05-24T11:46:23Z TRACE fred::protocol::codec] fred-wOdhQ0ju6H: Recv 0 bytes from 127.0.0.1:6379 (RESP2).
Got an int: 1!
```

**Additional context**
A custom command  is needed to return an array in response from redis.

**Comment by aembke** [comment]
This should be fixed in 7.x (starting at https://crates.io/crates/fred/7.0.0-rc.1). Unfortunately the fix was pretty invasive and required a major version bump. I'll leave this open until 7.0.0 is released though.

**Comment by aembke** [comment]
This should be fixed in [7.0.0](https://crates.io/crates/fred/7.0.0). Please let me know if you still see any issues here though.
